### ISSUE TITLE
a2530-msp432 implemantation and a2530 zigbee usage

### ISSUE THREAD

**Issue by cevatbostancioglu** [mweintra/ZigBee#1]
hello , i download your code , it's nice to work with boosterpack kits. i also try to implement to msp432 but i failed.
and you are using simple api , it's too bad choice , because people cannot usage module %60.
did you have a success solution for msp432 ? 
thanks.

**Comment by mweintra** [comment]
Hi Cevat,

Thanks for trying out the code! I haven't updated it for the msp432 yet (I
don't have this boosterpack). It doesn't use simple API, it actually is
built on an AF/ZDO interface. I'm not sure I understand module %60?

**Comment by cevatbostancioglu** [comment]
actually i'm try to say , if firmware use simple api , firmware can just use modules %60 skill.
and it's doesn't matter right now.
thanks for reply.
i think msp432 spi interface have a bug. i use msp432 p1_5,p3_0,p5_7,p1_7,p1_6,p2_5 pins but the firmware didn't work.

**Comment by dagarott** [comment]
Hello, first of all great job. Could you please, specify if there is any function in your code to get number of node that join the network, I mean, how many nodes form the mesh

Thanks and regards

**Comment by medsimo** [comment]
how can i do to connect 2 sensores node with 1 coordinator (gateway) ? i have many difficult to do this .... !!! help please.

**Comment by mweintra** [comment]
What kind of issues are you experiencing, specifically? Multiple sensors
should usually be able to connect to a Coordinator without any
modifications.

**Comment by medsimo** [comment]
The problem I have is, when I am going to connect two nodes to measure voltage and temperature, it turns out that one sends me the data well but the data is confused when I connect the second because they are repeated in the coordinator and I do not know why?
  With one I want to send external voltage and with another I want to send external frequency how how I could do it?

**Comment by medsimo** [comment]
Well, I also use an external voltage (read a source of 2.5 V, 3 V) and temperature sensor TMP006. How I do it. Some sample code or some help please. I am in a job end of career and I am very overwhelmed with these modules.

please mweintra !! 
regards.
Med :)

**Comment by mweintra** [comment]
How do you know the data received at the Coordinator is "repeated"? When
you receive a message do they have the same MAC or network address (you can
find this by using the .address() function)

As for the other questions not regarding this library you should try
posting it on http://forum.43oh.com/

**Comment by medsimo** [comment]
i don't know, because i can't use the MAC address (), Gives me bugs when I include the MAC address. 
can you give me your E.Mail, i will send you result and codes i have been using ?
![coordinador](https://cloud.githubusercontent.com/assets/23233776/22655487/994b016e-ec90-11e6-8ef6-197859ab7831.PNG)
![ed](https://cloud.githubusercontent.com/assets/23233776/22655489/994e6dfe-ec90-11e6-9655-3f6fff8e91f2.PNG)
![res 2](https://cloud.githubusercontent.com/assets/23233776/22655488/994cb75c-ec90-11e6-86c4-7d88c35f7c0c.png)
![res1](https://cloud.githubusercontent.com/assets/23233776/22655491/99743cdc-ec90-11e6-854a-ffde815476f8.png)
![router](https://cloud.githubusercontent.com/assets/23233776/22655490/994efa8a-ec90-11e6-9569-0795ccb51b23.PNG)

**Comment by mweintra** [comment]
It looks the coordinator is receiving data from both devices.

Add this to your main loop within the coordinator device:

Serial.println(ZigBee.address(),HEX);

This will tell you the address of the device sending you data. The data is
corrupted because of the misuse of the ZigBee.read() and write() functions.
Please take a look at the tutorials for proper usage.

**Comment by medsimo** [comment]
yes, the coordinator receive data from both devices. but ... data that it gives me are wrong, because I need the router to just send me the "counter" and the "phrase", and the end device, "voltage" and "temperature".

And it turns out that the data is sent to me twice !!! How can I solve it?
I'm really bothering a lot ... sorry :( but i need helps. 

![1](https://cloud.githubusercontent.com/assets/23233776/22690580/d0c94bb8-ed36-11e6-8cac-dd60b891bfcf.png)

**Comment by medsimo** [comment]
Okay, I have something, but I would like to know if these data are correct, because 18 degrees seem exaggerated to me in a room where the maximum is 13 degrees. The voltage does not know if it is good or not.

another question. How could you measure the temperature with the TMP006 of the CC2530 module and send it to the coordinator?
![1](https://cloud.githubusercontent.com/assets/23233776/22747556/a8087518-ee27-11e6-9fe7-a0444c27c408.png)

**Comment by mweintra** [comment]
Please post questions to issues other than this library to
http://forum.43oh.com/

**Comment by medsimo** [comment]
Ook !! I already have it in the forum, to see if someone answers me in a short time !!!

http://forum.43oh.com/topic/6082-energia-library-zigbee/

**Comment by medsimo** [comment]
http://forum.43oh.com/topic/10184-mweintra-zigbee-examples-for-cc2530/

any request !!

**Comment by medsimo** [comment]
Is there any way to be able to separate the data from the final device, and ROUTER?
That is, what I want to do is, put two boucles if with (DEVICE_ANNOUNCE), so that when the two are connected with the coordinator, the latter can read the data simultaneously, even the two of a command voltage and temperature, But they will be the recipients of the data.
I do not know if I get the DEVICE_ANNOUNCE I get nothing.

Any idea of ​​how the boucles have to be?
